docs/self_hosting/configuration/scale.mdx

```diff
@@ -1,41 +1,212 @@
+## Trace ingestion (write path)
 
-:::note
-Your exact usage pattern of the product may require more tuning of resources. If that is the case, please reach out to the LangSmith team for any questions.
+Services that play a large role in trace ingestion:
+
+- Platform backend service: Receives initial request to ingest traces and places traces on a Redis queue
+- Redis cache: Used to queue traces that need to be persisted
+- Queue service: Persists traces for querying
+- ClickHouse: Persistent storage used for traces
+
+Common usage that put load on the write path:
+
+- Ingesting traces via the Python or JavaScript LangSmith SDK
+- Ingesting traces via the `@traceable` wrapper
+- Submitting traces via the `/runs/multipart` endpoint
+
+When scaling up the write path (trace ingestion), it is helpful to monitor the four services/resources listed above. Here are some typical changes that can help increase performance of trace ingestion:
+
+- Give ClickHouse more resources (CPU and memory) if it is approaching resource limits.
+- Increase the number of platform-backend pods if ingest requests are taking long to respond.
+- Increase queue service pod replicas if traces are not being processed from Redis fast enough.
+- Use a larger Redis cache if you notice that the current Redis instance is reaching resource limits. This could also be a reason why ingest requests take a long time.
+
+## Trace querying (read path)
+
+Services that play a large role in querying traces:
+
+- Backend service: Receives the request and submits a query to ClickHouse to then respond to the request
+- ClickHouse: Persistent storage for traces. This is the main database that is queried when requesting trace info.
+
+Common usage that puts load on the read path:
+
+- Users on the frontend looking at tracing projects or individual traces
+- Scripts used to query for trace info
+- Hitting either the `/runs/query` or `/runs/<run-id>` api endpoints
+
+When scaling up the read path (trace querying), it is helpful to monitor the two services/resources listed above. Here are some typical changes that can help improve performance of trace querying:
+
+- Increase the number of backend service pods. This would be most impactful if backend service pods are reaching 1 core CPU usage.
+- Give ClickHouse more resources (CPU or Memory). ClickHouse can be very resource intensive, but it should lead to better performance.
+- Move to a [replicated ClickHouse cluster](./external_clickhouse#ha-replicated-clickhouse-cluster). Adding replicas of ClickHouse helps with read performance, but we recommend staying below 5 replicas (start with 3).
+
+For more precise guidance on how this translates to helm chart values, check out the examples below. If you are unsure why your LangSmith instance cannot handle a certain load pattern, please reach out to the LangChain team.
```

Comment:
Linking to the section, just in case additional content or the page changes and we (I) forget to update the language. Also, do we have an email or link for the "contact the LangChain team" part?

```suggestion
For more precise guidance on how this translates to helm chart values, refer to the examples the following [section](#example-langsmith-configurations-for-scale). If you are unsure why your LangSmith instance cannot handle a certain load pattern, contact the LangChain team.
```